filters/ema.hpp

```diff
@@ -38,71 +26,67 @@ class EMAFilter : public Consumer<In>, public Producer<double> {
     }
   }
 
-  ~EMAFilter() {}
-
-  virtual Try<Nothing> consume(const In& in) = 0;
-
-  void setAlpha(double _alpha) {
+  void setAlpha(double_t _alpha) {
     alpha = _alpha;
   }
 
-  double getAlpha() {
+  double_t getAlpha() {
     return alpha;
   }
 
- protected:
+  /*
+  * Calculate EMA and save needed values for next calculation.
+  */
+  double_t calculateEMA(double_t sample, double_t sampleTimestamp) {
+    if (uninitialized) {
+      prevEma = sample;
```

Comment:
You should use (*this). It's more clear way to see the side effects of this function.

Also, personally I would make this function change (*this) X-OR return result.
Not both at once.